wrapper/src/main/java/software/amazon/jdbc/plugin/customendpoint/CustomEndpointMonitorImpl.java

```diff
@@ -168,12 +171,30 @@ public void monitor() {
           TimeUnit.NANOSECONDS.sleep(sleepDuration);
         } catch (InterruptedException e) {
           throw e;
+        } catch (RdsException ex) {
+          LOGGER.log(Level.SEVERE,
+              Messages.get(
+                  "CustomEndpointMonitorImpl.exception",
+                  new Object[] {this.customEndpointHostSpec.getUrl()}), ex);
+
+          if (ex.isThrottlingException()) {
+            this.refreshRateNano *= 2; // Reduce the refresh rate.
```

Comment:
Will this number increase exponentially if the requests keep failing, will it cause the application to "hang"?

Reply:
No, it slows down a custom endpoint monitoring rate. Since such monitoring occurs in a separate thread there's no risk of "hanging" of the user application.

Reply:
This potentially always increases the refresh interval and never decreases it?

For a long-running process this just keeps going up and up; even if there are other processes that are causing a short-term rate throttling error.

Doing any type of rolling-restart of a service with many processes that all use the same custom endpoint will inevitably always start hitting the rate limiting and see these logger severe exceptions